File: backend/extract_pdf.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import pdfParse from 'pdf-parse';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const pdfPath = path.join(__dirname, '..', 'std guide.pdf');

async function extractPdf() {
    try {
        const dataBuffer = fs.readFileSync(pdfPath);
        const data = await pdfParse(dataBuffer);

        console.log('=== PDF TEXT CONTENT ===\n');
        console.log(data.text);
        console.log('\n=== END OF PDF ===');

        // Save to file for easier reading
        fs.writeFileSync(path.join(__dirname, 'pdf_content.txt'), data.text);
        console.log('\nSaved to backend/pdf_content.txt');
    } catch (error) {
        console.error('Error extracting PDF:', error);
    }
}

extractPdf();

File: backend/extract_pdf.py
import PyPDF2
import os

pdf_path = os.path.join(os.path.dirname(__file__), '..', 'std guide.pdf')
output_path = os.path.join(os.path.dirname(__file__), 'pdf_content.txt')

try:
    with open(pdf_path, 'rb') as file:
        reader = PyPDF2.PdfReader(file)
        text = ''
        for page in reader.pages:
            text += page.extract_text() + '\n\n--- PAGE BREAK ---\n\n'
        
        with open(output_path, 'w', encoding='utf-8') as out:
            out.write(text)
        
        print(text)
        print(f'\n\nSaved to {output_path}')
except Exception as e:
    print(f'Error: {e}')
